### DISCUSSION TITLE
Any way to override the default way of generating utilities? (generate-utility mixin)

### DISCUSSION THREAD

**Discussion by thiagomajesk** [twbs/bootstrap#33885] [General]
Some time ago I posted this issue: https://github.com/twbs/bootstrap/issues/32183 because I was confused about how Bootstrap was generating its utilities. At the time I was working on this project: https://github.com/thiagomajesk/bootilities, trying to mimic the [TailwindCSS](https://tailwindcss.com) utility conventions. 

In the end, I subscribed to Bootstrap conventions and ported some of the Tailwind utilities over to my project, started using it but I was not completely satisfied with the results. So I was wondering if it's possible to override the `generate-utility` mixin so users can apply their own conventions for generating utility classes. I think this would be great and would allow for great extensibility in the future. I don't know if @mdo will ever consider adding this level of extensibility to the framework, but a workaround could work too.

**Comment by mdo** [comment]
I'm open to anything if there's a way forward!

**Reply by thiagomajesk** [reply]
Wow, this is really reassuring to hear. I'm no Sass expert, but I'm researching to see how this could be accomplished. 
I started looking for dynamic imports, conditional imports, and things like that, but it seems that Sass has some hard limitations. Now I'm seeing documentation on the new modules stuff to see if there would be a native way to solve this.

Perhaps the easiest non-hacky way right now would be making the utility API a separate opt-in thing inside the project but I'm not sure I like this option either.  I know this is kinda possible today if you don't import the `bootstrap.scss` file directly, but it feels hacky to bypass the framework as an extension.

**Comment by thiagomajesk** [comment]
Hey @mdo! Since my last post here I got to know this new library called [Uniform CSS](https://uniformcss.com) (they went even further with my project idea)... Don't know if you have heard of it before, but I think it would be nice to look into it. Perhaps collaborating with them to use it as an engine for Bootstrap V6 utilities!? I don't know, but It would be really cool if both frameworks presented similar conventions in a way to facilitate component and utility usage in the same project.
